panic after upgrade from 2.1.1 to 2.3.0

Hello! after the upgrade of InfluxDB from 2.1.1 to 2.3.0 when I run this query

```
subintervals=2
topn=30
s=from(bucket:"prometheus")
|> range(start: v.timeRangeStart, stop: v.timeRangeStop)
|> filter(fn: (r) =>
  r["_measurement"] == "traffic_txrx_bytes_total")
|> aggregateWindow(every: duration(v: (int(v: v.timeRangeStop) - int(v: v.timeRangeStart)) / subintervals), fn: last, createEmpty: false)
|> derivative(unit: 1s)
|> group()
|> top(n:subintervals*topn)
|> sort(columns: ["_value"], desc: true)
|> unique(column:"ip")
|> top(n:topn)
|> findColumn(fn: (key) => true, column:"ip")

from(bucket:"prometheus")
|> range(start: v.timeRangeStart, stop: v.timeRangeStop)
|> filter(fn: (r) =>
  r["_measurement"] == "traffic_txrx_bytes_total" )
|> aggregateWindow(every: duration( v: int( v: $__interval)*10), fn: last, createEmpty: false)
|> derivative(unit: 1s)
|> filter(fn: (r) => contains(value: r.ip, set: s))
|> keep(columns:["_time","_value","app","pod_name"])

```
I get the following panic error

```
panic: runtime error: index out of range [0] with length 0 stacktrace=goroutine 82101 [running]:
runtime/debug.Stack()
        /go/src/runtime/debug/stack.go:24 +0x65
github.com/influxdata/flux/execute.(*poolDispatcher).recover(0xc09844eb60)
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/recover.go:53 +0x205
panic({0x3255060, 0xc0411f4588})
        /go/src/runtime/panic.go:838 +0x207
github.com/apache/arrow/go/v7/arrow/array.(*Float64).Value(...)
        /go/pkg/mod/github.com/apache/arrow/go/v7@v7.0.0/arrow/array/numeric.gen.go:225
github.com/influxdata/flux/internal/arrowutil.FloatCompareDesc(0x7ef93e9f1980?, 0x29574?, 0x470a70?, 0x43667d?)
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/internal/arrowutil/compare.gen.go:177 +0x16d
github.com/influxdata/flux/internal/arrowutil.CompareDesc({0x33c52b8?, 0xc0941dbc80?}, {0x33c52b8?, 0xc0cb0ceb40?}, 0x121ef91?, 0x121efce?)
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/internal/arrowutil/compare.gen.go:59 +0x85
github.com/influxdata/flux/stdlib/universe.(*sortTableMergeHeap).Less(0xc0cb0d4300, 0x128be6a?, 0x128bba5?)
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/stdlib/universe/sort.go:304 +0x119
container/heap.down({0x33bc1d0, 0xc0cb0d4300}, 0x0, 0x2)
        /go/src/container/heap/heap.go:112 +0x109
container/heap.Init({0x33bc1d0, 0xc0cb0d4300})
        /go/src/container/heap/heap.go:46 +0x53
github.com/influxdata/flux/stdlib/universe.(*sortTableMergeHeap).Table(0xc0cb0d4300, 0x3c, {0x33afb08, 0xc08b252690})
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/stdlib/universe/sort.go:346 +0x169
github.com/influxdata/flux/stdlib/universe.(*sortLimitTransformation).Aggregate(0xc094190820, {{{0x33cc520, 0xc0941917c0}, {0xc0944b8a00, 0xd, 0xd}, {0xc0cb0c4ea0, 0xd, 0xd}}}, {0x31feb20, ...}, ...)
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/stdlib/universe/sort_limit.go:87 +0x3a5
github.com/influxdata/flux/execute.(*aggregateTransformation).processChunk(0xc0941a0ae0, {{{0x33cc520, 0xc0941917c0}, {0xc0944b8a00, 0xd, 0xd}, {0xc0cb0c4ea0, 0xd, 0xd}}})
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/aggregate.go:85 +0x165
github.com/influxdata/flux/execute.(*aggregateTransformation).ProcessMessage(0x31cf360?, {0x33b6468?, 0xc0cb0d4660})
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/aggregate.go:74 +0x29c
github.com/influxdata/flux/execute.(*transportTransformationAdapter).processChunk(...)
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/transport.go:757
github.com/influxdata/flux/execute.(*transportTransformationAdapter).Process.func1({0x33ca690?, 0xc0941dbd40?})
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/transport.go:744 +0x275
github.com/influxdata/flux/execute.(*consecutiveTransportTable).Do.func1({0x33ca690, 0xc0941dbd40})
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/transport.go:567 +0x489
github.com/influxdata/flux/execute/table.(*BufferedTable).Do(0x10?, 0xc0cb0be7b0)
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/table/buffered.go:69 +0x19b
github.com/influxdata/flux/execute.(*consecutiveTransportTable).Do(0xc0cb0be780, 0xc0cb09f700)
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/transport.go:549 +0x87
github.com/influxdata/flux/execute.(*transportTransformationAdapter).Process(0xc0941e02c0, {0x23, 0xd1, 0x92, 0xc9, 0x73, 0xf8, 0x5e, 0x4a, 0xa2, ...}, ...)
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/transport.go:741 +0x390
github.com/influxdata/flux/execute.(*transportTransformationAdapter).ProcessMessage(0xc0941e02c0, {0x33b64a0?, 0xc0cb09f680})
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/transport.go:722 +0x13e
github.com/influxdata/flux/execute.(*consecutiveTransport).processMessage(0xc084482a50, {0x33b64a0, 0xc0cb09f680})
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/transport.go:292 +0xca
github.com/influxdata/flux/execute.(*consecutiveTransport).processMessages(0xc084482a50, {0x33b56d8?, 0xc0941da380?}, 0xa)
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/transport.go:246 +0xe5
github.com/influxdata/flux/execute.(*poolDispatcher).doWork(0xc09844eb60, {0x33b56d8, 0xc0941da380})
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/dispatcher.go:157 +0x19f
github.com/influxdata/flux/execute.(*poolDispatcher).run(0xc09844eb60, {0x33b56d8, 0xc0941da380})
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/dispatcher.go:135 +0x4f
github.com/influxdata/flux/execute.(*poolDispatcher).Start.func1()
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/dispatcher.go:75 +0x86
created by github.com/influxdata/flux/execute.(*poolDispatcher).Start
        /go/pkg/mod/github.com/influxdata/flux@v0.171.0/execute/dispatcher.go:71 +0x48
```

this is InfluxDB v.2.3.0 running on Red Hat Enterprise Linux release 8.6 (Ootpa)

```
InfluxDB v2.3.0+SNAPSHOT.090f681737
Server: [090f681](https://github.com/influxdata/influxdb/tree/090f681737)
Frontend: [a2bd1f3](https://github.com/influxdata/ui/tree/a2bd1f3)
```

--- a/libflux/go/libflux/buildinfo.gen.go
+++ b/libflux/go/libflux/buildinfo.gen.go
@@ -576,7 +576,7 @@ var sourceHashes = map[string]string{
 	"stdlib/universe/simple_max_test.flux":                                                        "a63b3f530e4d81451e3c71a1abeea50cff02e743c9313ab3ffd5bc3b3ce9ad2e",
 	"stdlib/universe/skew_test.flux":                                                              "7782d41c563c77ba9f4176fa1b5f4f6107e418b7ea301e4896398dbcb514315a",
 	"stdlib/universe/sort2_test.flux":                                                             "1d2043c0d0b38abb8dc61fc1baa6d6052fae63fea55cc6e67fd1600911513bdb",
-	"stdlib/universe/sort_limit_test.flux":                                                        "5ae162efb25cb25c3f18b871ab30bfd5e827acb329fd92d8926011658df8bd82",
+	"stdlib/universe/sort_limit_test.flux":                                                        "80a7a25384b86db401ffd6fd2ea58c8e4b0646f522fa121e92a00793c0a28039",
 	"stdlib/universe/sort_rules_test.flux":                                                        "0770ae42e99b04167ca5bef8340a310b224baf1ba1928997273de9663b64684a",
 	"stdlib/universe/sort_test.flux":                                                              "f69ebb5972762078e759af3c1cd3d852431a569dce74f3c379709c9e174bfa31",
 	"stdlib/universe/spread_test.flux":                                                            "1ddf25e4d86b6365da254229fc8f77bd838c24a79e6a08c9c4c50330ace0a6a3",

--- a/stdlib/universe/sort.go
+++ b/stdlib/universe/sort.go
@@ -339,7 +339,7 @@ func (s *sortTableMergeHeap) ValueLen() int {
 
 func (s *sortTableMergeHeap) Table(limit int, mem memory.Allocator) (flux.Table, error) {
 	if s.ValueLen() == 0 {
-		// Degenerate case where there are no rows to merge sort
+		// Degenerate case where there are no rows to merge sort.
 		for len(s.items) > 0 {
 			s.Pop()
 		}

--- a/stdlib/universe/sort_limit.go
+++ b/stdlib/universe/sort_limit.go
@@ -80,6 +80,13 @@ func (s *sortLimitTransformation) Aggregate(chunk table.Chunk, state interface{}
 
 	}
 
+	// If the chunk is empty, ignore this chunk.
+	// We still return the merge heap though as we still need to recognize that
+	// the group key/schema exist.
+	if chunk.Len() == 0 {
+		return mh, true, nil
+	}
+
 	if err := s.appendChunk(mh, chunk, mem); err != nil {
 		return nil, false, err
 	}

--- a/stdlib/universe/sort_limit_test.flux
+++ b/stdlib/universe/sort_limit_test.flux
@@ -170,3 +170,29 @@ testcase sort_limit_multi_successor {
 
     testing.diff(got: got, want: want)
 }
+
+testcase sort_limit_empty_chunk {
+    got =
+        array.from(
+            rows: [
+                {_time: 2022-01-11T00:00:00Z, t0: "aa", _value: 12.0},
+                {_time: 2022-01-11T00:00:00Z, t0: "ab", _value: 10.0},
+                {_time: 2022-01-11T00:00:00Z, t0: "ba", _value: 18.0},
+                {_time: 2022-01-11T00:00:00Z, t0: "bb", _value: 4.0},
+            ],
+        )
+            |> group(columns: ["t0"])
+            |> filter(fn: (r) => r.t0 =~ /b/, onEmpty: "keep")
+            |> group()
+            |> top(n: 2)
+
+    want =
+        array.from(
+            rows: [
+                {_time: 2022-01-11T00:00:00Z, t0: "ba", _value: 18.0},
+                {_time: 2022-01-11T00:00:00Z, t0: "ab", _value: 10.0},
+            ],
+        )
+
+    testing.diff(got: got, want: want)
+}

